rust/interpreter/src/evaluate.rs

```diff
@@ -112,6 +133,37 @@ pub fn evaluate(emit: &EmitResult) -> Result<JSValue, EvalError> {
                 continue;
             }
 
+            Opcode::GetGName => {
+                let atom = emit.read_atom(pc + 1);
+                stack.push(global.borrow().get(atom));
+            }
+
+            Opcode::GImplicitThis => {
+                stack.push(JSValue::Object(global.clone()));
+            }
+
+            Opcode::Call => {
+                let len = emit.read_u16(pc + 1);
+
+                let mut arguments = Vec::new();
+                for _ in 0..len {
+                    let argument = stack.pop().ok_or(EvalError::EmptyStack)?;
+                    arguments.insert(0, argument);
+                }
```

Comment:
You could check the stack depth once, then use `stack.split_off(stack.len() - argc)`.

Or `&stack[args_start..]` to pass a reference to the slice, and `stack.truncate(args_start - 2);` afterwards.